django-stubs/test/client.pyi

```diff
@@ -155,7 +156,8 @@ class _RequestFactory(Generic[_T]):
 class RequestFactory(_RequestFactory[WSGIRequest]): ...
 
 @type_check_only
-class _AsyncRequestFactory(_RequestFactory[_T]): ...
+class _AsyncRequestFactory(_RequestFactory[_T]):
+    defaults: HTTPScope  # type: ignore[assignment]
```

Comment:
Wait! I didn't think this `HTTPScope` thing through.

`HTTPScope` isn't using `total=False` or `NotRequired`, so if some user tries to use the `defaults` attribute, they are required to fill **all** fields. 😨

What are our options? I think we can't change the totality of `HTTPScope TypedDict` on our side. So we could duplicate the whole type on our side, or we could just revert back to `dict[str, Any]`. Sorry about that.

Reply:
I think that's actually desired?

A full `HTTPScope` is always required I think. As such, if you reassign defaults (e.g. `defaults =`) you need to populate all fields. But you could modify some of them, e.g. `.defaults["client"] = ...`

Reply:
Haha, now I'm reading this wrong. You're right, since Django updates the scope with `defaults`, all of them aren't required.

Reply:
`asgiref` only uses `NotRequired` for `state`. But we want `total=False`.

I'm thinking we should duplicate the type.